core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -338,24 +345,51 @@ private[spark] class MemoryStore(
     * RDDs that don't fit into memory that we want to avoid).
     *
     * @param blockId the ID of the block we are freeing space for, if any
-    * @param space the size of this block
-    * @return the amount of memory (in bytes) freed by eviction
+    * @param requestedSpace the amount of memory which should be freed via block eviction
+    * @return the total size (in bytes) of the blocks evicted from the MemoryStore. This memory is
+    *         not released with the memory manager, so it's the caller's responsibility to perform
+    *         the appropriate bookkeeping calls.
```

Comment:
I don't understand why this should be the case. I get that we don't want to acquire the lock for each block we drop, but can't we just release all the memory at the end of this method? We still only acquire 1 lock and now the caller doesn't have to worry about forgetting to release memory.